kcp/run.sh

```diff
@@ -0,0 +1,236 @@
+  oc apply -f "$SCRIPT_DIR/manifests/pln-svc-manager.yaml" --wait
+}
+
+
+
+get_context(){
+  KUBECONFIG="$1"
+  local namespace="$2"
+  local sa="$3"
+  local target="$4"
+  local current_context
+  local sa_context
+  current_context="$(oc config current-context)"
+  sa_context="$KCP_ENV-${sa//-manager/}"
+
+  if ! which jq &> /dev/null; then
+    echo "[ERROR] Install jq"
+    exit 1
+  fi
+  token_secret="$(oc get sa "$sa" -n "$namespace" -o json | jq -r '.secrets[].name | select(. | test(".*token.*"))')"
+  current_context="$(oc config current-context)"
+  current_cluster="$(oc config view -o jsonpath="{.contexts[?(@.name==\"$current_context\")].context.cluster}")"
+
+  oc config set-credentials "$sa" --token="$(oc get secret "$token_secret" -n "$namespace" -o jsonpath="{.data.token}" | base64 -d)" &> /dev/null
+  oc config set-context "$sa_context" --user="$sa" --cluster="$current_cluster" &> /dev/null
+  oc config use-context "$sa_context" &> /dev/null
+  oc config view --flatten --minify > "$target"
+  oc config use-context "$current_context" &> /dev/null
+}
+
+
+
+create_kcp_contexts(){
+  echo -n "  - Generating '$KUBECONFIG_KCP_PLNSVC': "
+  get_context "$KUBECONFIG" pln-svc pln-svc-manager "$KUBECONFIG_KCP_PLNSVC"
+  echo "OK"
+
+  # Check if this can be deprecated when ArgoCD runs on KCP instead of the Pipeline cluster
+  echo -n "  - Generating '$KUBECONFIG_KCP_ARGOCD': "
+  get_context "$KUBECONFIG" kube-system argocd-manager "$KUBECONFIG_KCP_ARGOCD"
+  echo "OK"
+}
+
+
+
+register_kcp_to_argocd(){
+  # Register the KCP cluster to ArgoCD
+  export KUBECONFIG="$KUBECONFIG_MERGED"
+  local argcocd_cluster_name="kcp"
+  echo -n "  - Register cluster to ArgoCD as '$argcocd_cluster_name': "
+  argocd cluster add "$KCP_ENV-argocd" --name="$argcocd_cluster_name" --service-account argocd-manager --upsert --yes >/dev/null
+  echo "OK"
+}
+
+
+
+configure_kcp_cluster(){
+  echo "[KCP cluster]"
+  setup_kcp_service_accounts
+  create_kcp_contexts
+  register_kcp_to_argocd
+  echo
+}
+
+
+
+install_apps(){
+  echo "[Install applications]"
+  export KUBECONFIG="$KUBECONFIG_PLNSVC_CLUSTERADMIN"
+  local gitops_path
+  gitops_path="$(cd "$SCRIPT_DIR/../gitops"; pwd)"
+
+  echo "  - on KCP: "
+  # Create kcp-kubeconfig secrets for event listener and interceptors so that they can talk to KCP
+  KUBECONFIG="$KUBECONFIG_KCP" oc apply -f "$gitops_path/triggers/triggers-crds/base/namespace.yaml" --wait &>/dev/null
+  oc create secret generic kcp-kubeconfig --from-file "$KUBECONFIG_KCP_PLNSVC" --dry-run=client -o yaml | KUBECONFIG="$KUBECONFIG_KCP" oc apply -f - --wait &>/dev/null
+  KUBECONFIG="$KUBECONFIG_KCP" oc apply -f "https://raw.githubusercontent.com/tektoncd/pipeline/v0.32.0/config/100-namespace.yaml" --wait &>/dev/null
+  oc create secret generic kcp-kubeconfig -n tekton-pipelines --from-file "$KUBECONFIG_KCP_PLNSVC" --dry-run=client -o yaml | KUBECONFIG="$KUBECONFIG_KCP" oc apply -f - --wait &>/dev/null
+
```

Comment:
The way KUBECONFIG is
- sometimes exported
- sometimes set in the command line
makes it more difficult to read the script.

Proposal: reusing the environment variables defined at the start of the script and having kubeconfig specified with each command, example:
`oc apply --kubeconfig=${KUBECONFIG_KCP} -f "https://raw.githubusercontent.com/tektoncd/pipeline/v0.32.0/config/100-namespace.yaml" --wait`

Reply:
Commands are already quite long, so I used `oc_kcp` and `oc_plnsvc` function to wrap the oc calls with the expected KUBECONFIG.
FYI, the `--kubeconfig` option did not seem to work with the `kcp` plugin, which is why I went with setting the env variable.